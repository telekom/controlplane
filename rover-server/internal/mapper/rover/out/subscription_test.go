// Copyright 2025 Deutsche Telekom IT GmbH
//
// SPDX-License-Identifier: Apache-2.0

package out

import (
	"github.com/gkampitakis/go-snaps/snaps"
	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"
	roverv1 "github.com/telekom/controlplane/rover/api/v1"

	"github.com/telekom/controlplane/rover-server/internal/api"
)

var _ = Describe("Subscription Mapper", func() {
	Context("MapApiSubscription", func() {
		It("must map ApiSubscription correctly", func() {
			input := &apiSubscription

			output := mapApiSubscription(input)

			Expect(output).ToNot(BeNil())
			snaps.MatchSnapshot(GinkgoT(), output)
		})

		It("must handle empty ApiSubscription input", func() {
			input := &roverv1.ApiSubscription{}

			output := mapApiSubscription(input)

			Expect(output).ToNot(BeNil())
			snaps.MatchSnapshot(GinkgoT(), output)
		})
	})

	Context("mapSubscription", func() {
		It("must map an ApiSubscription correctly", func() {
			input := GetApiSubscription(&apiSubscription)
			output := &api.Subscription{}

			err := mapSubscription(&input, output)

			Expect(err).To(BeNil())
			snaps.MatchSnapshot(GinkgoT(), output)
		})

		It("must map an EventSubscription correctly", func() {
			input := GetEventSubscription(&eventSubscription)
			output := &api.Subscription{}

			err := mapSubscription(&input, output)

			Expect(err).To(BeNil())
			snaps.MatchSnapshot(GinkgoT(), output)
		})

		It("must return an error if Discriminator fails", func() {
			input := &roverv1.Subscription{}
			output := &api.Subscription{}

			err := mapSubscription(input, output)

			Expect(err).ToNot(BeNil())
			Expect(err.Error()).To(ContainSubstring("unknown subscription type"))
			snaps.MatchSnapshot(GinkgoT(), output)
		})
	})

})
