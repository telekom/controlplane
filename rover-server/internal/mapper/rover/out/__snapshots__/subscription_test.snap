
[Subscription Mapper MapApiSubscription must map ApiSubscription correctly - 1]
api.ApiSubscription{
    BasePath: "/eni/distr/v1",
    Failover: api.Failover{},
    Security: api.Security{},
    Type:     "",
}
---

[Subscription Mapper MapApiSubscription must handle empty ApiSubscription input - 1]
api.ApiSubscription{}
---

[Subscription Mapper mapSubscription must map an ApiSubscription correctly - 1]
&api.Subscription{
    union: {0x7b, 0x22, 0x62, 0x61, 0x73, 0x65, 0x50, 0x61, 0x74, 0x68, 0x22, 0x3a, 0x22, 0x2f, 0x65, 0x6e, 0x69, 0x2f, 0x64, 0x69, 0x73, 0x74, 0x72, 0x2f, 0x76, 0x31, 0x22, 0x2c, 0x22, 0x74, 0x79, 0x70, 0x65, 0x22, 0x3a, 0x22, 0x61, 0x70, 0x69, 0x22, 0x7d},
}
---

[Subscription Mapper mapSubscription must map an EventSubscription correctly - 1]
&api.Subscription{
    union: {0x7b, 0x22, 0x64, 0x65, 0x6c, 0x69, 0x76, 0x65, 0x72, 0x79, 0x54, 0x79, 0x70, 0x65, 0x22, 0x3a, 0x22, 0x22, 0x2c, 0x22, 0x65, 0x76, 0x65, 0x6e, 0x74, 0x54, 0x79, 0x70, 0x65, 0x22, 0x3a, 0x22, 0x74, 0x65, 0x73, 0x74, 0x2d, 0x65, 0x76, 0x65, 0x6e, 0x74, 0x22, 0x2c, 0x22, 0x70, 0x61, 0x79, 0x6c, 0x6f, 0x61, 0x64, 0x54, 0x79, 0x70, 0x65, 0x22, 0x3a, 0x22, 0x22, 0x2c, 0x22, 0x74, 0x79, 0x70, 0x65, 0x22, 0x3a, 0x22, 0x65, 0x76, 0x65, 0x6e, 0x74, 0x22, 0x7d},
}
---

[Subscription Mapper mapSubscription must return an error if Discriminator fails - 1]
&api.Subscription{}
---
